src/qililab/platform/platform.py

```diff
@@ -574,13 +574,34 @@ def execute_qprogram(  # pylint: disable=too-many-locals
         calibration: Calibration | None = None,
         debug: bool = False,
     ) -> QProgramResults:
-        """Execute a QProgram using the platform instruments.
+        """Execute a :class:`.QProgram` using the platform instruments.
+
+        |
+
+        **The execution is done in the following steps:**
+
+        1. Compile the QProgram.
+        2. Run the compiled QProgram.
+        3. Acquire the results.
+
+        |
+
+        **The execution is done in the following way:**
+
+        - If all the buses in the :class:`.QProgram` are associated with Qblox modules, the compilation is done using the :class:`.QbloxCompiler`.
+        - If all the buses in the :class:`.QProgram` are associated with Quantum Machines Clusters, the compilation is done using the :class:`.QuantumMachinesCompiler`.
+        - If the buses in the :class:`.QProgram` are associated with a mixture of Qblox modules and Quantum Machines Clusters, the compilation is not supported.
```

Comment:
I would replace this part (as I think that it is not saying much) for a more specific comment regarding the specific operations inside Qblox and QM

```suggestion
        **execute_qprogram operates with two different clusters independently:**

        - For buses assosiated with Qblox modules, the compilation is done using the :class:`.QbloxCompiler`. This compiler compiles the Qprogram in Q1ASM into multiple sequencers based on each bus, uploads and executes the sequences, acquires the results and resets the settings by desyncing each bus.
        - For buses assosiated with Quantum Machines Clusters, the compilation is done using the :class:`.QuantumMachinesCompiler`. This compiler transforms the Qprogram in QUA, the compilation language for Quantum Machines, runs the job created by the QUA execution and returns the results ordering them by bus.
```
This can be definetly improved as each person has different ways to explain themselves
BTW I really like the :class:`.something`.